### PR TITLE
Use fetch and remove request [5.x]

### PR DESCRIPTION
**Pull request by ayusharma**
**Type:** Refactor

The following has been addressed in the PR:

*  There is a related issue #1807 
*  Unit or Functional tests are included in the PR

**Description:**

### CODE REVIEW

File: packages/hooks/test/request.spec.ts [hunk 1]
```diff
@@ -27,72 +25,37 @@ describe('Notifications:: notifyRequest', () => {
...
+    const args = [{ message: 'bad data' }, 'notify service has thrown an error: @{message}'];
 
-    await expect(notification.notifyRequest(options, content)).rejects.toEqual(API_ERROR.BAD_DATA);
+    await expect(notification.notifyRequest(url, {}, content)).rejects.toThrowError();
     expect(logger.logger.error).toHaveBeenCalledWith(...args);
   });
 
   test('when notification is successfully delivered', async () => {
-    jest.doMock('request', () => (options, resolver) => {
+    jest.doMock('node-fetch', () => (options, resolver) => {
```
Comment (juanpicado) [inline packages/hooks/test/request.spec.ts:45]: You can remove `options, resolver` is not being used

File: packages/hooks/test/request.spec.ts [hunk 2]
```diff
@@ -27,72 +25,37 @@ describe('Notifications:: notifyRequest', () => {
...
+        ok: true
       };
 
-      resolver(null, response, response.body);
+     return Promise.resolve(response);
     });
 
     const notification = require('../src/notify-request');
     const infoArgs = [{ content }, 'A notification has been shipped: @{content}'];
     const debugArgs = [{ body: 'Successfully delivered' }, ' body: @{body}'];
```
Comment (juanpicado) [inline packages/hooks/test/request.spec.ts:56]: `debugArgs` is not being used.

File: packages/server/test/api/index.spec.ts [hunk 3]
```diff
@@ -47,38 +47,43 @@ describe('endpoint unit test', () => {
...
-    const binPath = require.resolve('verdaccio/bin/verdaccio');
-    const storePath = path.join(__dirname, '/mock/store');
-    mockRegistry = await mockServer(mockServerPort, { storePath, silence: true }).init(binPath);
-    done();
+      }, 'api.spec.yaml', __dirname);
+
+      app = await endPointAPI(configForTest);
+      const binPath = require.resolve('verdaccio/bin/verdaccio');
+      const storePath = path.join(__dirname, '/mock/store');
+      console.log('DEBUG 🔥', storePath);
```
Comment (ayusharma) [inline packages/server/test/api/index.spec.ts:73]: **NOTE: REMOVE THIS**

**Comment by juanpicado** [comment]
Another option with retry https://www.npmjs.com/package/make-fetch-happen

**Comment by juanpicado** [comment]
@ayusharma you might want to have some inspiration here https://github.com/verdaccio/monorepo/pull/367

**Comment by ayusharma** [comment]
Thanks @juanpicado 😀

**Comment by juanpicado** [comment]
Closing this, `master` is completely different  now.

**Pull request closed without merging**